### PR TITLE
Replace flake8 and black with ruff

### PR DESCRIPTION
**Pull request by jacalvo**
https://github.com/astral-sh/ruff

Apart from being faster, it should add additional linters we were missing like `isort`.

I'm also removing `requirements.txt` as it was redundant with `pyproject.toml`.

Finally some versions are slightly updated due to running `poetry lock`.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-  rev: 7.1.0
-  hooks:
-  - id: flake8
-    args: ['--max-line-length', '110']
-    additional_dependencies: [flake8-bugbear]
+    # Run the linter.
+    - id: ruff
+      args: [ --fix ]
+    # Run the formatter.
+    - id: ruff-format
```
Comment (figarocorso) [inline .pre-commit-config.yaml:10]: Remove chatgpt comments 😂

Comment (jacalvo) [reply .pre-commit-config.yaml:10]: They are not ChatGPT comments, it's the "official" file from upstream, check https://github.com/astral-sh/ruff-pre-commit :)

Comment (figarocorso) [reply .pre-commit-config.yaml:10]: Shit! You win, this time =)

**Review by tembleking — APPROVED** [review]
Added also changes for the flake.nix file. Thanks jos!

**Pull request merged**